src/block/Sugarcane.php

```diff
@@ -98,6 +98,12 @@ public function ticksRandomly() : bool{
 
 	public function onRandomTick() : void{
 		if(!$this->getSide(Facing::DOWN)->hasSameTypeId($this)){
+			if(!$this->isSupportedByWater()) {
+				$world = $this->position->getWorld();
+				$world->useBreakOn($this->position, createParticles: true);
```

Comment:
There is no need to create an additional `$world` variable. You can inline this.